pkg/crc/machine/kubeconfig.go

```diff
@@ -0,0 +1,109 @@
+package machine
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+)
+
+func WriteKubeconfig(config ClusterConfig) error {
+	dir, err := ioutil.TempDir("", "crc")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(dir)
+	dev, err := writeContext(dir, config, "developer", "developer", "crc-dev")
+	if err != nil {
+		return err
+	}
+	admin, err := writeContext(dir, config, "kubeadmin", config.KubeAdminPass, "crc-admin")
+	if err != nil {
+		return err
+	}
+
+	rootConfig := filepath.Join(constants.GetHomeDir(), ".kube", "config")
+	context := currentContext(rootConfig)
+	if err := mergeKubeconfig(admin, dev, rootConfig); err != nil {
+		return err
+	}
+	if context != "" {
+		return restoreCurrentContext(context, rootConfig)
+	}
+	return nil
+}
+
+func restoreCurrentContext(context string, rootConfig string) error {
+	// #nosec G204
+	restoreCmd := exec.Command(ocBinary(), "config", "use-context", strings.TrimSpace(string(context)))
+	restoreCmd.Stderr = os.Stderr
+	restoreCmd.Env = []string{fmt.Sprintf("KUBECONFIG=%s", rootConfig)}
+	return restoreCmd.Run()
+}
+
+func ocBinary() string {
+	return filepath.Join(constants.CrcOcBinDir, constants.OcBinaryName)
+}
+
+func mergeKubeconfig(admin string, dev string, rootConfig string) error {
+	// #nosec G204
+	mergeCmd := exec.Command(ocBinary(), "config", "view")
+	mergeCmd.Stderr = os.Stderr
+	if _, err := os.Stat(rootConfig); os.IsNotExist(err) {
+		mergeCmd.Env = []string{fmt.Sprintf("KUBECONFIG=%s:%s", admin, dev)}
+	} else {
+		mergeCmd.Env = []string{fmt.Sprintf("KUBECONFIG=%s:%s:%s", admin, dev, rootConfig)}
+	}
+	out, err := mergeCmd.Output()
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(rootConfig, out, 0600)
+}
+
+func currentContext(rootConfig string) string {
+	if _, err := os.Stat(rootConfig); os.IsNotExist(err) {
+		return ""
+	}
+	// #nosec G204
+	currentCtxCmd := exec.Command(ocBinary(), "config", "current-context")
+	currentCtxCmd.Stderr = os.Stderr
+	currentCtxCmd.Env = []string{fmt.Sprintf("KUBECONFIG=%s", rootConfig)}
+	context, err := currentCtxCmd.Output()
```

Comment:
For this you can use https://github.com/code-ready/crc/blob/master/pkg/os/exec.go#L54 one also use `--kubeconfig` parameter for oc command instead of adding to env.

Reply:
It's weird but config merge doesn't work with `--kubeconfig`. I had to keep the env variable.

``` 
$ kubectl --kubeconfig c1:c2 config view
apiVersion: v1
clusters: []
contexts: []
current-context: ""
kind: Config
preferences: {}
users: []

$ KUBECONFIG=c1:c2 kubectl config view  
apiVersion: v1
clusters:
- cluster:
    insecure-skip-tls-verify: true
    server: https://api.crc.testing:6443
...

```

Reply:
@guillaumerose Ah yes, https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#merging-kubeconfig-files explains it.

Reply:
I refactored this, it should be better.